service/awsconfig/v10/resource/cloudformation/update.go

```diff
@@ -18,41 +18,43 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	updateStackInput, err := toUpdateStackInput(updateChange)
+	stackStateToUpdate, err := toStackState(updateChange)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	if updateStackInput.StackName != nil && *updateStackInput.StackName != "" {
+	if stackStateToUpdate.Name != "" {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "updating the guest cluster main stack")
 
-		// Fetch the etcd volume information.
-		etcdVolume := true
-		persistentVolume := false
-		volumes, err := r.ebs.ListVolumes(customObject, etcdVolume, persistentVolume)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-		if len(volumes) != 1 {
-			return microerror.Maskf(executionFailedError, "there must be 1 volume for etcd, got %d", len(volumes))
-		}
-		vol := volumes[0]
-
-		// First shutdown the instances and wait for it to be stopped. Then detach
-		// the etcd volume without forcing.
-		force := false
-		shutdown := true
-		wait := true
-		for _, a := range vol.Attachments {
-			err := r.ebs.DetachVolume(ctx, vol.VolumeID, a, force, shutdown, wait)
+		if stackStateToUpdate.ShouldUpdate && !stackStateToUpdate.ShouldScale {
```

Comment:
This is something I discovered now. I have really no idea how this was not caught before because I tested this actually for two days. Problem here is that when scaling the master is still brought down. We make this conditional now so we only bring the master down in case we process an update. On scaling we ignore the shutdown and detach magic.

Reply:
Good that this is caught now.